src/Workspaces/Remote/ServiceHub/Host/TestUtils.cs

```diff
@@ -204,30 +203,35 @@ private static async Task AppendAssetMapAsync(this Project project, Dictionary<C
             }
         }
 
-        private static HashSet<Checksum> Flatten(ChecksumWithChildren checksums)
+        private static HashSet<Checksum> Flatten(SolutionStateChecksums checksums)
         {
             var set = new HashSet<Checksum>();
-            set.AppendChecksums(checksums);
+            checksums.AddAllTo(set);
+            return set;
+        }
 
+        private static HashSet<Checksum> Flatten(ProjectStateChecksums checksums)
+        {
+            var set = new HashSet<Checksum>();
+            checksums.AddAllTo(set);
             return set;
         }
 
-        public static void AppendChecksums(this HashSet<Checksum> set, ChecksumWithChildren checksums)
+        private static HashSet<Checksum> Flatten(DocumentStateChecksums checksums)
+        {
+            var set = new HashSet<Checksum>();
+            checksums.AddAllTo(set);
+            return set;
+        }
+
+        public static void AppendChecksums(this HashSet<Checksum> set, ChecksumCollection checksums)
         {
             set.Add(checksums.Checksum);
 
             foreach (var child in checksums.Children)
             {
-                if (child is Checksum checksum)
-                {
-                    if (checksum != Checksum.Null)
-                        set.Add(checksum);
-                }
-
-                if (child is ChecksumCollection collection)
-                {
-                    set.AppendChecksums(collection);
-                }
+                if (child != Checksum.Null)
+                    set.Add(child);
             }
```

Comment:
just preserving oldb ehavior.  but i have a feeling most of these helpers are not necessary.